packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/EnumTypeValue.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Generator.CSharp
+{
+    public class EnumTypeValue
+    {
+        public EnumTypeValue(string name, FieldDeclaration field, object value)
+        {
+            Name = name;
+            Field = field;
+            Value = value;
+        }
+
+        public string Name { get; }
```

Comment:
I hate this but we still need to have the `Name` here.

In extensible enums, the name of the values, for example: `A, B, C`, will show up twice in fields and in properties like this:
```csharp
public readonly partial struct Foo
{
    private const string AValue = "A";
    private const string BValue = "B";
    private const string CValue = "C";

    public Foo A {get;} = new Foo(AValue);
    public Foo B {get;} = new Foo(BValue);
    public Foo C {get;} = new Foo(CValue);
}
```
When we are building the properties, we still need to have the original name for the value because the name of the field will be `AValue` and we want `A` here.
Also I think it is fragile if we just trim off the `Value` suffix because in the future we might get complicated cases and the field name must not be `AValue`, for example, someone could define an extensible enum like this:
```typespec
union Foo extends string {
    string,
    "A",
    "AValue"
}
```
And in this case, we must change the name of `AValue` because it collides with a name of the property which we cannot change. For example, we will have to have this:
```csharp
public readonly partial struct Foo 
{
    private const string AValue1 = "A";
    private const string AValueValue = "AValue";
    
    public Foo A {get;} = new Foo(AValue1);
    public Foo AValue {get;} = new Foo(AValueValue);
}
```